### PR TITLE
Support customer managed keys for encryption

### PR DESCRIPTION
**Pull request by KimForss**
## Problem
Add support for customer managed keys for disk encryption

## Solution
This PR provides customer managed keys support to VMs of sapsystem and deployer

Add a property in json in the options section
"disk_encryption_set_id" : "/subscriptions/xxxxxxxxxx/resourceGroups/PROTO-NOEU-MGMT-INFRASTRUCTURE/providers/Microsoft.Compute/diskEncryptionSets/PROTO-NOEU-KEY_SET",

## Tests
Create a disk encryption key in keyvault and att the property to the input json

## Notes
<Additional comments for the PR>

### CODE REVIEW

**Comment by YunziZhangAzure** [comment]
Thanks for the PR! Do we consider supporting customer managed keys for disk encryption for deployer VM?

File: deploy/terraform/terraform-units/modules/sap_system/common_infrastructure/vm-anchor.tf [hunk 1]
```diff
@@ -36,9 +36,10 @@ resource "azurerm_linux_virtual_machine" "anchor" {
   disable_password_authentication = true
 
   os_disk {
-    name                 = format("%s%s%s%s", local.prefix, var.naming.separator, local.anchor_virtualmachine_names[count.index], local.resource_suffixes.osdisk)
-    caching              = "ReadWrite"
-    storage_account_type = "Standard_LRS"
+    name                   = format("%s%s%s%s", local.prefix, var.naming.separator, local.anchor_virtualmachine_names[count.index], local.resource_suffixes.osdisk)
+    caching                = "ReadWrite"
+    storage_account_type   = "Premium_LRS"
```
Comment (YunziZhangAzure) [inline deploy/terraform/terraform-units/modules/sap_system/common_infrastructure/vm-anchor.tf:41]: Same question as above.

File: deploy/terraform/terraform-units/modules/sap_system/anydb_node/vm-observer.tf [hunk 2]
```diff
@@ -50,9 +50,10 @@ resource "azurerm_linux_virtual_machine" "observer" {
   disable_password_authentication = true
 
   os_disk {
-    name                 = format("%s%s%s%s", local.prefix, var.naming.separator, local.observer_virtualmachine_names[count.index], local.resource_suffixes.osdisk)
-    caching              = "ReadWrite"
-    storage_account_type = "Standard_LRS"
+    name                   = format("%s%s%s%s", local.prefix, var.naming.separator, local.observer_virtualmachine_names[count.index], local.resource_suffixes.osdisk)
+    caching                = "ReadWrite"
+    storage_account_type   = "Premium_LRS"
```
Comment (YunziZhangAzure) [inline deploy/terraform/terraform-units/modules/sap_system/anydb_node/vm-observer.tf:55]: Why does storage account type changed from Standard_LRS to premium_LRS?

File: deploy/terraform/terraform-units/modules/sap_system/common_infrastructure/vm-anchor.tf (hunk 1 above)
Comment (KimForss) [reply deploy/terraform/terraform-units/modules/sap_system/common_infrastructure/vm-anchor.tf:41]: All Production VMs should use Premium Disks, it is better to default to that

File: deploy/terraform/terraform-units/modules/sap_system/anydb_node/vm-observer.tf (hunk 2 above)
Comment (KimForss) [reply deploy/terraform/terraform-units/modules/sap_system/anydb_node/vm-observer.tf:55]: All Production VMs should use Premium Disks, it is better to default to that

**Comment by KimForss** [comment]
> Thanks for the PR! Do we consider supporting customer managed keys for disk encryption for deployer VM?

Yes - Just added that

**Review by YunziZhangAzure — APPROVED** [review]

**Pull request merged**